node/modules/retrieval.go

```diff
@@ -58,3 +70,74 @@ func HandleRetrievalTransports(lc fx.Lifecycle, l *lp2pimpl.TransportsListener)
 		},
 	})
 }
+
+func NewLoadBalancer(cfg *config.Boost) func(h host.Host) (*loadbalancer.LoadBalancer, error) {
+	return func(h host.Host) (*loadbalancer.LoadBalancer, error) {
+		peerConfig := map[peer.ID][]protocol.ID{}
+		if cfg.Dealmaking.BitswapPeerID != "" {
+			bsPeerID, err := peer.Decode(cfg.Dealmaking.BitswapPeerID)
+			if err != nil {
+				return nil, err
+			}
+			peerConfig[bsPeerID] = []protocol.ID{
+				network.ProtocolBitswap,
+				network.ProtocolBitswapNoVers,
+				network.ProtocolBitswapOneOne,
+				network.ProtocolBitswapOneZero,
+			}
+		}
+		return loadbalancer.NewLoadBalancer(h, peerConfig)
+	}
+}
+
+func HandleLoadBalancer(lc fx.Lifecycle, lb *loadbalancer.LoadBalancer) {
+	lc.Append(fx.Hook{
+		OnStart: func(ctx context.Context) error {
+			log.Debug("starting load balancer")
+			lb.Start(ctx)
+			return nil
+		},
+		OnStop: func(context.Context) error {
+			log.Debug("stopping load balancer")
+			return lb.Close()
+		},
+	})
+}
+
+func NewSetBitswapPeerIDFunc(r lotus_repo.LockedRepo, lb *loadbalancer.LoadBalancer) (dtypes.SetBitswapPeerIDFunc, error) {
+	return func(p peer.ID) (err error) {
+		err = mutateCfg(r, func(cfg *config.Boost) {
+			cfg.Dealmaking.BitswapPeerID = peer.Encode(p)
+		})
+		if err != nil {
+			return
+		}
+		peerConfig := map[peer.ID][]protocol.ID{}
+		peerConfig[p] = []protocol.ID{
+			network.ProtocolBitswap,
+			network.ProtocolBitswapNoVers,
+			network.ProtocolBitswapOneOne,
+			network.ProtocolBitswapOneZero,
+		}
+		err = lb.UpdatePeerConfig(peerConfig)
+		return
+	}, nil
+}
+
+func NewGetBitswapPeerIDFunc(r lotus_repo.LockedRepo) (dtypes.GetBitswapPeerIDFunc, error) {
+	return func() (p peer.ID, err error) {
+		var pString string
+		err = mutateCfg(r, func(cfg *config.Boost) {
+			pString = cfg.Dealmaking.BitswapPeerID
+		})
+		if err != nil {
+			return
+		}
+		if pString == "" {
+			err = fmt.Errorf("no bitswap peer id set")
```

Comment:
For simplicity I'd suggest we just return "" here instead of an error. That way we don't need to do string matching across the API boundary to detect the string "no bitswap peer id set"